### PR TITLE
feat: add calculate_tax_loss_harvesting MCP tool

### LINKED ISSUE #25: feat: タックスロスハーベスティング計算ツールの追加
## 概要
損失確定可能なポジションの特定と節税効果の計算、ウォッシュセールルールチェックを行うツールを追加する。

### calculate_tax_loss_harvesting
- 現在含み損のあるポジションをリストアップ
- 損失確定による節税効果を計算（マレーシア税制考慮）
- ウォッシュセールルール（30日以内の同銘柄再購入）の自動チェック
- 代替銘柄の提案（例: INDA → NDIA.L）

## Acceptance Criteria
- [ ] `ib_sec_mcp/mcp/tools/ib_portfolio.py` に `calculate_tax_loss_harvesting` を追加
- [ ] ウォッシュセール30日ルールのチェックロジック実装
- [ ] Decimal精度で損益計算
- [ ] アイルランド籍代替ETFの提案ロジック
- [ ] ユニットテストを追加

## 価値
年末の税務対策の自動化、損失確定のタイミング判断支援。

## Wave
Wave 4（Wave 3と並列実行可能）

### PR DESCRIPTION
**Pull request by knishioka**
## Summary

Implements #25 - Adds a `calculate_tax_loss_harvesting` MCP tool for identifying tax loss harvesting opportunities across IB portfolio positions.

- Identifies all positions with unrealized losses and calculates potential tax savings
- Implements wash sale rule (30-day window) detection for both forward-looking (current positions) and backward-looking (historical trades)
- Suggests Ireland-domiciled alternative ETFs (e.g., INDA -> NDIA.L) to maintain market exposure while harvesting losses
- Supports configurable tax rate (default 30%, set to 0 for jurisdictions like Malaysia)
- Classifies holding periods as short-term (<365 days) or long-term

## Changes

| File | Change |
|------|--------|
| `ib_sec_mcp/mcp/tools/ib_portfolio.py` | Added `calculate_tax_loss_harvesting` tool (+238 lines) |
| `tests/mcp/test_ib_portfolio.py` | Added 17 unit tests with XML fixtures (+467 lines) |

## Test plan

- [x] 17 unit tests covering all acceptance criteria
- [x] Loss position identification and gain exclusion
- [x] Decimal precision for all financial calculations
- [x] Wash sale risk detection (30-day window)
- [x] Historical wash sale warnings
- [x] Alternative ETF suggestions (INDA -> NDIA.L, EEM -> IEEM.L, etc.)
- [x] Zero tax rate handling (Malaysia jurisdiction)
- [x] Input validation (invalid tax_rate, out-of-range values)
- [x] Empty portfolio edge case
- [x] Result structure and sorting verification
- [x] Full test suite passes (123/123)
- [x] ruff format/check clean
- [x] mypy strict mode clean

Closes #25

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: ib_sec_mcp/mcp/tools/ib_portfolio.py [hunk 1]
```diff
@@ -1058,6 +1059,241 @@ async def analyze_consolidated_portfolio(
...
+            raise ValidationError("No accounts found in data")
+
+        account_list = list(accounts.values())
+        if account_index >= len(account_list):
+            raise ValidationError(
+                f"account_index {account_index} out of range (0-{len(account_list) - 1})"
+            )
+
+        account = account_list[account_index]
+        today = date.today()
```
[the remark this answers is not in the document]
Comment (knishioka) [reply ib_sec_mcp/mcp/tools/ib_portfolio.py]: Fixed in c7a9e29. Replaced `date.today()` with `to_date` (analysis period end date) for all holding period and wash sale calculations. This ensures historical analysis produces correct results regardless of when the tool is run.

File: ib_sec_mcp/mcp/tools/ib_portfolio.py [hunk 2]
```diff
@@ -1058,6 +1059,241 @@ async def analyze_consolidated_portfolio(
...
+                                        "message": (
+                                            f"Potential wash sale: {symbol} sold at loss on "
+                                            f"{trade.trade_date}, rebought on "
+                                            f"{future_trade.trade_date} "
+                                            f"({days_between} days later)"
+                                        ),
+                                    }
+                                )
+                                break
+
```
[the remark this answers is not in the document]
Comment (knishioka) [reply ib_sec_mcp/mcp/tools/ib_portfolio.py]: Fixed in c7a9e29. Added backward-looking wash sale check that iterates backwards from the loss sale to detect purchases within 30 days before the sale, in addition to the existing forward-looking check. This covers both directions per IRS wash sale rule.

File: ib_sec_mcp/mcp/tools/ib_portfolio.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (knishioka) [reply ib_sec_mcp/mcp/tools/ib_portfolio.py]: Fixed in c7a9e29. Replaced `date.today()` with `to_date` for all holding period and wash sale window calculations. Historical analysis now produces correct results based on the analyzed period, not the current date.
